Update Relay SDK version

fjgal opened the pull request:
Update the relay SDK version.

* accommodates breaking change to `Configuration.BasePath`
* takes advantage of new `AllocationIdBytes` field - avoid Guid to byte array conversion issues (byte ordering)
* removes obsolete workaround for connection data length
* update default relay endpoint to production
* updated dependency on `com.unity.services.authentication` and related code

Tested using `testproject`

fjgal committed:
update to relay sdk v0.0.1-preview.4

fjgal committed:
remove workarounds

- use byte array representation of allocation id
- connection data truncation is not longer needed
- set default relay endpoint as production

lpmaurice commented on testproject/Packages/manifest.json:
```diff
@@ -6,10 +6,11 @@
     "com.unity.ide.vscode": "1.2.3",
     "com.unity.multiplayer.mlapi": "file:../../com.unity.multiplayer.mlapi",
     "com.unity.multiplayer.transport.utp": "file:../../com.unity.multiplayer.transport.utp",
+    "com.unity.transport": "file:../../../com.unity.transport",
```
We can't assume that customers will have this package installed, nor that it would be at this exact location. The transport package is already included in the `com.unity.multiplayer.transport.utp` package, so I would prefer we keep this as the source of truth.

It does mean that while the UTP PR is not approved and the actual package updated on the registry, devs will need to make a change manually in the project though.

fjgal replied:
100% i accidentally committed that change. Thanks for catching it!

fjgal committed:
updated auth to v0.5.0

- updated core and authentication versions
- updated relay version (compatible with latest core)
- updated authentication calls

fjgal committed:
remove project dependency on utp

lpmaurice commented on testproject/Packages/packages-lock.json:
```diff
@@ -116,42 +117,39 @@
     },
     "com.unity.services.relay": {
-      "version": "0.0.1-preview.4",
+      "version": "file:../../../com.unity.services.relay/com.unity.services.relay",
```
Is this an accidental commit?

lpmaurice commented on testproject/Packages/packages-lock.json:
```diff
@@ -116,42 +116,39 @@
     },
     "com.unity.services.relay": {
-      "version": "0.0.1-preview.3",
+      "version": "file:../../../com.unity.services.relay/com.unity.services.relay",
```
I think this file, `packages-lock.json`, also needs to be reverted. It contains path to local repos, which makes me think it was also commited by mistake.

fjgal committed:
revert accidental changes to package-lock

lpmaurice reviewed (approved):
LGTM
